pygmt/src/filter1d.py

```diff
@@ -0,0 +1,144 @@
+"""
+filter1d - Time domain filtering of 1-D data tables
+"""
+import warnings
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile, build_arg_string, fmt_docstring, use_alias
+
+
+@fmt_docstring
+@use_alias(
+    E="end",
+    F="filter",
+)
+def filter1d(data, output_type="pandas", outfile=None, **kwargs):
+    r"""
+    Time domain filtering of 1-D data tables.
+
+    Filter a general time domain for multiple column time
+    series data. The user specifies which column is the time (i.e., the
+    independent variable). The fastest operation
+    occurs when the input time series are equally spaced and have no gaps or
+    outliers and the special options are not needed.
+    Read a table and output as a :class:`numpy.ndarray`,
+    :class:`pandas.DataFrame`, or ASCII file.
+
+    Full option list at :gmt-docs:`filter1d.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    filter : str
+        **type**\ *width*\ [**+l**\|\ **u**].
+        Sets the filter **type**. Choose among convolution and non-convolution
+        filters. Append the filter code followed by the full filter
+        *width* in same units as time column. By default, this
+        performs a low-pass filtering; append **+h** to select high-pass
+        filtering. Some filters allow for optional arguments and a modifier.
+
+        Available convolution filter types are:
+
+        (**b**) Boxcar: All weights are equal.
+
+        (**c**) Cosine Arch: Weights follow a cosine arch curve.
+
+        (**g**) Gaussian: Weights are given by the Gaussian function.
+
+        (**f**) Custom: Instead of *width* give name of a one-column file
+        with your own weight coefficients.
+
+        Non-convolution filter types are:
+
+        (**m**) Median: Returns median value.
+
+        (**p**) Maximum likelihood probability (a mode estimator): Return
+        modal value. If more than one mode is found we return their average
+        value. Append **+l** or **+u** if you rather want
+        to return the lowermost or uppermost of the modal values.
+
+        (**l**) Lower: Return the minimum of all values.
+
+        (**L**) Lower: Return minimum of all positive values only.
+
+        (**u**) Upper: Return maximum of all values.
+
+        (**U**) Upper: Return maximum of all negative values only.
```

Comment:
<img width="807" alt="image" src="https://user-images.githubusercontent.com/3974108/164971173-dbd92be6-dbb0-4c1e-98dc-589d8d3a3b9a.png">

I feel the current documentation is not easy to read, so I reformat the documentation like below:

```suggestion
        Available convolution filter types are:

        - (**b**) Boxcar: All weights are equal.
        - (**c**) Cosine Arch: Weights follow a cosine arch curve.
        - (**g**) Gaussian: Weights are given by the Gaussian function.
        - (**f**) Custom: Instead of *width* give name of a one-column file
          with your own weight coefficients.

        Non-convolution filter types are:

        - (**m**) Median: Returns median value.
        - (**p**) Maximum likelihood probability (a mode estimator): Return
          modal value. If more than one mode is found we return their average
          value. Append **+l** or **+u** if you rather want
          to return the lowermost or uppermost of the modal values.
        - (**l**) Lower: Return the minimum of all values.
        - (**L**) Lower: Return minimum of all positive values only.
        - (**u**) Upper: Return maximum of all values.
        - (**U**) Upper: Return maximum of all negative values only.
```